src/Codegen/Generator.hack

```diff
@@ -200,9 +210,16 @@ class Field {
     protected function getArgumentInvocationStrings(): vec<string> {
         $invocations = vec[];
         foreach ($this->reflection_method->getParameters() as $index => $param) {
+            $type_text = $param->getTypeText();
+            // TODO: can we pass down dynamic context this way too?
+            if ($type_text === 'HH\classname<Slack\GraphQL\Introspection\IntrospectableSchema>') {
```

Comment:
We need to pass the schema down to inspect it. I added some special handling here if a function takes a schema object we'll pass it the schema. I thought this ended up being pretty nice and could let itself to supporting custom context objects. I'm open to other ways we can do this though.

Reply:
Yeah this is a nice way to support context objects if needed, but I'm not sure we need it for the schema...couldn't these fields just do `new Schema()` instead of having the schema passed in? (or `Schema::get()` if we make it a singleton) The schema doesn't have any state that we'd need to pass around, or does it?

Reply:
> couldn't these fields just do new Schema() instead of having the schema passed in?

I guess the fear is that `Schema` doesn't exist until after codegen, so we'll be writing classes which reference a type that the type-checker can't necessarily resolve? Seems odd to have non-generated code reference generated code. But maybe I'm missing something?